### PR TITLE
Add envelope check to DistanceOp::computeMinDistance

### PR DESCRIPTION
**Pull request by dbaston**
Reduces `GEOSDistance` runtime by 70% on a test set of polygons.

### CODE REVIEW

**Comment by dr-jts** [comment]
Nice!  That works in JTS as well - will make the change.

**Comment by dr-jts** [comment]
@dbaston I also tried adding a short-circuit after the first `for` statement, checking the distance from the segment envelope to the other line envelope.  That seemed to produce a further 50% reduction in my test data.  What do you think?

**Comment by dbaston** [comment]
Confirmed, @dr-jts . Now looking at overall 90% improvement in `GEOSDistance`.

**Comment by dr-jts** [comment]
Excellent.

It's worth noting that using `IndexedFacetDistance` on the US-Europe test is still 3x faster than `DistanceOp` with the new optimizations.  But this certainly makes the improvement with indexing a lot less significant.

File: src/operation/distance/DistanceOp.cpp [hunk 1]
```diff
@@ -447,7 +447,7 @@ DistanceOp::computeMinDistance(
 
         Envelope segEnv0(p00, p01);
 
-        if (segEnv0.distance(lineEnv1) > minDistance) {
+        if (segEnv0.distanceSquared(lineEnv1) > minDistance*minDistance) {
```
Comment (dr-jts) [inline src/operation/distance/DistanceOp.cpp]: Is this more performant?

Comment (dbaston) [reply src/operation/distance/DistanceOp.cpp]: Modestly but measurably.

**Pull request merged**